src/software/simulated_tests/terminating_validation_functions/robot_has_orientation_validation.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include "software/simulated_tests/validation/validation_function.h"
+#include "software/world/world.h"
+
+/**
+ * Checks if the robot represented by robot_id has arrived at the destination at
+ * any point in the test
```

Comment:
Maybe consider replacing "destination" with "orientation" or similar to make it clear it's checking for the orientation and not coordinates of the robot.

Reply:
whoops good catch